presentation/src/main/java/me/androidbox/presentation/BusbyTaskyApplication.kt

```diff
@@ -1,7 +1,45 @@
 package me.androidbox.presentation
 
 import android.app.Application
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.content.Context
+import android.os.Build
 import dagger.hilt.android.HiltAndroidApp
 
 @HiltAndroidApp
-class BusbyTaskyApplication : Application()
\ No newline at end of file
+class BusbyTaskyApplication : Application() {
+
+    override fun onCreate() {
+        super.onCreate()
+
+        createNotificationChannel(generateChannels())
+    }
+
+    private fun createNotificationChannel(listOfChannel: Map<String, String>) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            listOfChannel.map { mapOfChannel ->
+                val notificationManager = getNotificationManager()
+                val notificationChannel = NotificationChannel(
+                    mapOfChannel.key,
+                    mapOfChannel.value,
+                    NotificationManager.IMPORTANCE_HIGH
+                )
+
+                notificationManager.createNotificationChannel(notificationChannel)
+            }
+        }
+    }
+
+    private fun getNotificationManager(): NotificationManager {
+        return this.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+    }
+
+    private fun generateChannels(): Map<String, String> {
+        return mapOf(
+            "Notifications for EVENTS" to "event",
+            "Notifications for TASKS" to "task",
+            "Notifications for REMINDERS" to "reminders"
+        )
+    }
```

Comment:
@philipplackner Is this the best way to create 3 separate channels

Reply:
Here, read comment above